xds/internal/client/client.go

```diff
@@ -401,9 +401,19 @@ type SecurityConfig struct {
 	RequireClientCert bool
 }
 
+// ClusterType is the type of cluster from a received CDS response.
+type ClusterType int
+
+const (
+	eds = iota
```

Comment:
and `eds ClusterType = iota`, so you won't need to cast later.
Like this: https://github.com/grpc/grpc-go/blob/master/xds/internal/client/client.go#L152